src/project-type/workspace-instance.ts

```diff
@@ -0,0 +1,34 @@
+import { WorkspaceApi, WorkspaceImpl } from "@sap/project-api";
+import { BasWorkspaceApi } from "../../types/api";
+
+let basWSAPI: BasWorkspaceApi;
+
+export function getBasWorkspaceAPI(): BasWorkspaceApi {
+  if (!basWSAPI) {
+    const workspaceImpl = new WorkspaceImpl();
```

Comment:
I think we will need to separate the concerns in this module so it would be easily testable.

lets discuss offline